src/root/DepositManager.ts

```diff
@@ -0,0 +1,69 @@
+import Web3 from 'web3'
+import { Contract } from 'web3/types'
+import { address, BigNumber, SendOptions } from '../types/Common'
+import DepositManagerArtifact from '../artifacts/DepositManager.json'
+
+export default class DepositManager {
+  private _depositManagerContract: Contract
+  private _defaultOptions: SendOptions
+
+  constructor(_depositManagerContractAddress: address, _web3: Web3, _defaultOptions?: SendOptions) {
+    this._depositManagerContract = new _web3.eth.Contract(DepositManagerArtifact.abi, _depositManagerContractAddress)
+    this._defaultOptions = _defaultOptions || {}
+  }
+
+  setDepositManagerAddress(_depositManagerContractAddress: address) {
+    this._depositManagerContract.options.address = _depositManagerContractAddress
```

Comment:
Why do we require this `this._depositManagerContract.options.address`?

any specific reason for simply not using `this._depositManagerContract.address`